src/workerd/api/html-rewriter.c++

```diff
@@ -394,13 +397,19 @@ void Rewriter::output(const char* buffer, size_t size, void* userdata) {
   rewriter.outputImpl(buffer, size);
 }
 
-void Rewriter::outputImpl(const char* buffer, size_t size) {
-  if (isPoisoned()) {
-    // Handlers disabled due to exception or running in a destructor.
+void Rewriter::writeFromBuffer(kj::Maybe<kj::Array<char>> maybeData) {
+  auto size = maybeData.map([](kj::Array<char>& b) { return b.size(); }).orDefault(0);
+  auto builder = kj::heapArrayBuilder<char>(writeBufferPos + size);
```

Comment:
It doesn't hurt, but I don't think this needs to be a builder, since `char` is trivial.

Reply:
Specifically, this could be
```cpp
  auto bufferCopy = kj::heapArray<char>(totalSize);
  std::copy(writeBuffer.begin(), writeBuffer.begin() + writeBufferPos, bufferCopy.begin());
  KJ_IF_MAYBE(data, maybeData) {
    std::copy(data->begin(), data->end(), writeBuffer.begin() + writeBufferPos);
  }
  writeBufferPos = 0;
```
But now that I write that out, I think your builder-based implementation is obviously more readable, so let's pretend I didn't make this suggestion.